hax/treemakers/posrec.py

```diff
@@ -0,0 +1,163 @@
+                's1_patterns_zoom_factor', 1),
+            adjust_to_qe=qes[self.tpc_channels],
+            default_errors=(self.pax_config['DEFAULT']['relative_qe_error'] +
+                            self.pax_config['DEFAULT']['relative_gain_error'])
+        )
+
+        # For s1_aft variable
+        # Don't yell at me for hardcoding the filename into hax because it was
+        # also hardcoded in pax. Just kicking the can.
+        #aftmap_filename = utils.data_file_name('s1_aft_xyz_XENON1T_06Mar2017.json')
+        aftmap_filename = utils.data_file_name('s1_aft_xyz_XENON1T_20170808.json')
+        self.aft_map = InterpolatingMap(aftmap_filename)
+        self.low_pe_threshold=10
+        #hax.minitrees.Treemaker.__init__(self)
+        # load trained NN models
+        nn_model_json = utils.data_file_name('tensorflow_nn_pos_XENON1T_20171211.json')
+        json_file_nn = open(nn_model_json, 'r')
+        loaded_model_json = json_file_nn.read()
+        json_file_nn.close()
+        loaded_nn_model = model_from_json(loaded_model_json)
+        weights_file = utils.data_file_name('tensorflow_nn_pos_weights_XENON1T_20171211.h5')
+        loaded_nn_model.load_weights(weights_file)
+        self.nn_tensorflow = loaded_nn_model
+        self.list_bad_pmts = [1, 2, 12, 26, 34, 62, 65, 79, 86, 88, 102, 118, 130, 134, 135, 139, 148, 150, 152, 162, 178,
+                         183, 190, 198, 206, 213, 214, 234, 239, 244]
+        self.ntop_pmts = 127 # How to get this automatically?
+
+    def get_data(self, dataset, event_list=None):
+
+        data, _ = hax.minitrees.load_single_dataset(dataset, ['Corrections',
+                                                              'Fundamentals'])
+        self.x = data.x_3d_nn.values
+        self.y = data.y_3d_nn.values
+        self.z = data.z_3d_nn.values
+
+        self.indices = list(data.event_number.values)
+
+        return hax.minitrees.TreeMaker.get_data(self, dataset, event_list)
+
+    def extract_data(self, event):
+
+        event_data = {
+            "s1_pattern_fit_hax": None,
+            "s1_area_fraction_top_probability_hax": None
+        }
+
+        # We first need the positions. This minitree is only valid when loading
+        # Corrections since you need that to get the corrected positions
+        if not len(event.interactions):
+            return event_data
+        e = event.event_number
+        try:
+            i=self.indices.index(e)
+        except:
+            return event_data
+        interaction = event.interactions[0]
+        s1 = event.peaks[interaction.s1]
+        s2 = event.peaks[interaction.s2]
+
+        # Position reconstruction based on NN from TensorFlow
+        s2apc = np.array(list(s2.area_per_channel))
+        s2apc_clean = []
+        for i, s2_t in enumerate(s2apc):
+            if i not in self.list_bad_pmts and i < self.ntop_pmts:
+                s2apc.clean.append(s2_t)
+        s2acp_clean.np.assaray(s2acp_clean)
```

Comment:
acp -> apc?

Reply:
Fixed in https://github.com/XENON1T/hax/pull/174/commits/c8d44aa448b7571906c6e5d2a2e645be2d870640